Map Json string to Object instead of HashMap

AlpriElse opened the pull request:
Resolves https://github.com/HubSpot/jinjava/issues/457
Related to https://hubspotdev.slack.com/archives/C2YH119N2/p1592854855031000

This pr changes `jinjava.lib.filter.FromJsonFilter` to return a `Object` instead of a `HashMap` in order to also support deserializing JSON array strings [(see issue)]( https://github.com/HubSpot/jinjava/issues/457).

Now, we are able to use array JSON strings:
```
// template
{% test = "[one, two, three]" | fromjson %}
{{  test[0] }}

// output
one

/* Before this would raise an InvalidInputException */
```

mattcoley commented on src/test/java/com/hubspot/jinjava/lib/filter/FromJsonFilterTest.java:
```diff
@@ -46,21 +48,73 @@ public void itFailsWhenParameterIsNotString() {
-  private void checkedNestJson(HashMap<String, Object> node) {
-    assertThat(node.get("first")).isEqualTo(Arrays.asList(1, 2, 3));
+  @Test
+  public void itReadsEmptyJsonArrayString() {
```
I think a better option for these tests is to render a jinjava template directly. See https://github.com/HubSpot/jinjava/blob/master/src/test/java/com/hubspot/jinjava/lib/filter/MinusTimeFilterTest.java#L37 as an example. That way we can make sure the API from Jinjava perspective is the same: for example: `{% set obj = test|fromjson %} {{ obj.property }}`

AlpriElse commented:
@mattcoley I reworked the testing to check rendered jinjava. Also, I removed the old tests because we were testing implementation and not behavior... which isn't ideal. Also, it looks like using `Object` instead of `JsonNode` gives us the intended functionality we want.

One thing I did notice that you should be aware of is the behavior of using the `pprint` filter with this:
```
// template
{% set obj = "{a: 100, b: [1, 2, 3, 4]}" %}
{{ obj | fromjson | pprint }}

// output on Design Manager:
(String: {a: 100, b: [1, 2, 3, 4]})

// output now:
(PyMap: {a=100, b=[1, 2, 3, 4]})
```

AlpriElse commented on .travis.yml:
```diff
@@ -21,6 +21,6 @@ notifications:
   - pip install --user codecov
 after_success:
-  - mvn validate -e
+  - travis_wait mvn validate -e
```
I've been getting build failures because some packages take a while to install.
https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

AlpriElse commented:
Build is succeeding in Blazar, going to go ahead with merge.
